tools/restatectl/src/util.rs

```diff
@@ -8,13 +8,64 @@
 // the Business Source License, use of this software will be governed
 // by the Apache License, Version 2.0.
 
+use std::fmt::{self, Display};
+
 use tonic::transport::Channel;
 
 use restate_cli_util::CliContext;
 use restate_core::network::net_util::create_tonic_channel;
-use restate_types::net::AdvertisedAddress;
+use restate_types::{logs::metadata::ProviderConfiguration, net::AdvertisedAddress};
 
 pub fn grpc_channel(address: AdvertisedAddress) -> Channel {
     let ctx = CliContext::get();
     create_tonic_channel(address, &ctx.network)
 }
+
+pub fn write_default_provider<W: fmt::Write>(
+    w: &mut W,
+    depth: usize,
+    provider: &ProviderConfiguration,
+) -> Result<(), fmt::Error> {
+    let title = "Logs Provider";
+    match provider {
+        #[cfg(any(test, feature = "memory-loglet"))]
+        ProviderConfiguration::InMemory => {
+            write_leaf(w, depth, true, title, "in-memory")?;
+        }
+        ProviderConfiguration::Local => {
+            write_leaf(w, depth, true, title, "local")?;
+        }
+        #[cfg(feature = "replicated-loglet")]
+        ProviderConfiguration::Replicated(config) => {
+            write_leaf(w, depth, true, title, "replicated")?;
+            let depth = depth + 2;
```

Comment:
No need for indenting too much.
```suggestion
            let depth = depth + 1;
```